examples/decomposition/plot_faces_decomposition.py

```diff
@@ -57,183 +70,245 @@ def plot_gallery(title, images, n_col=n_col, n_row=n_row, cmap=plt.cm.gray):
-            n_iter=50,
-            batch_size=3,
-            random_state=rng,
-            positive_dict=True,
-        ),
-        True,
-    ),
-    (
-        "Dictionary learning - positive code",
-        decomposition.MiniBatchDictionaryLearning(
-            n_components=15,
-            alpha=0.1,
-            n_iter=50,
-            batch_size=3,
-            fit_algorithm="cd",
-            random_state=rng,
-            positive_code=True,
-        ),
-        True,
-    ),
-    (
-        "Dictionary learning - positive dictionary & code",
-        decomposition.MiniBatchDictionaryLearning(
-            n_components=15,
-            alpha=0.1,
-            n_iter=50,
-            batch_size=3,
-            fit_algorithm="cd",
-            random_state=rng,
-            positive_dict=True,
-            positive_code=True,
-        ),
-        True,
-    ),
-]
-
-
-# #############################################################################
-# Plot a sample of the input data
+# %%
+# Let’s take a look at our data. Gray color indicates negative values,
+# white indicates positive values.
+
+plot_gallery("Faces from dataset", faces_centered[:n_components])
+
+# %%
+# Decomposition
+# --------------
+#
+# Initialise different estimators for decomposition and fit each
+# of them on all images and plot some results. Each estimator extracts
+# 6 components as vectors :math:`h \in \mathbf{R}^{4096}`.
+# We just displayed these vectors in human-friendly visualisation as 64x64 pixel images.
+#
+# Read more in the :ref:`User Guide <decompositions>`.
+
+# %%
+# Eigenfaces - PCA using randomized SVD
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Linear dimensionality reduction using Singular Value Decomposition (SVD) of the data
+# to project it to a lower dimensional space.
+#
+#
+# .. note::
+#
+#     The Eigenfaces estimator, via the :py:mod:`sklearn.decomposition.PCA`,
+#     also provides a scalar `noise_variance_` (the mean of pixelwise variance)
+#     that cannot be displayed as an image.
+
+# %%
+pca_estimator = decomposition.PCA(
+    n_components=n_components, svd_solver="randomized", whiten=True
+)
+pca_estimator.fit(faces_centered)
+plot_gallery(
+    "Eigenfaces - PCA using randomized SVD", pca_estimator.components_[:n_components]
+)
+
+# %%
+# Non-negative components - NMF
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Estimate non- negative original data as production of two non-negative matrices.
+
+# %%
+nmf_estimator = decomposition.NMF(n_components=n_components, tol=5e-3)
+nmf_estimator.fit(faces)  # original non- negative dataset
+plot_gallery("Non-negative components - NMF", nmf_estimator.components_[:n_components])
+
+# %%
+# Independent components - FastICA
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Independent component analysis separates a multivariate vectors into additive
+# subcomponents that are maximally independent.
+
+# %%
+ica_estimator = decomposition.FastICA(n_components=n_components, whiten=True)
```

Comment:
We can remove the `FutureWarning`.

```suggestion
ica_estimator = decomposition.FastICA(
    n_components=n_components, whiten='arbitrary-variance'
)
```

Regarding the `ConvergenceWarning`, could you increase the number of iterations (or maybe increase the tolerance) such that we don't get this warning anymore.

Reply:
Fixed. increased `max_iter` and `tol`